3.5/test/run

```diff
@@ -168,7 +168,7 @@ test_application() {
 
 # Since we built the candidate image locally, we don't want S2I attempt to pull
 # it from Docker hub
-s2i_args="--force-pull=false"
+s2i_args="--incremental=true --force-pull=false"
```

Comment:
i don't think this is sufficient to test this new functionality.  it really needs to:

1) build the app image
2) build the app image again passing --incremental and confirm (via the log output, perhaps) that the artifacts from (1) were reused successfully instead of being re-downloaded.